src/modules/cubedos-file_server-api.ads

```diff
@@ -54,91 +55,240 @@ package CubedOS.File_Server.API is
+   procedure Open_Request_Encode
+      (Sender_Address : Message_Address;
+      Receiver_Address : Message_Address;
+      Request_ID : Request_ID_Type;
+      Mode : Mode_Type;
+      Name : String;
+      Result : out Message_Record;
+      Priority : System.Priority := System.Default_Priority)
+   with
+      Global => (Proof_In => Mailbox_Metadata),
+      Pre => true
+         and then (0 < Name'Length and Name'Length <= XDR_Size_Type'Last - 12)
+         and then Receiver_Address.Module_ID = This_Module
+         and then Receives(Receiver_Address.Module_ID, Open_Request_Msg),
+      Post => Message_Manager.Message_Type(Result) = Open_Request_Msg
+       and Message_Manager.Receiver_Address(Result) = Receiver_Address;
+
+   procedure Send_Open_Request
+      (Sender : Module_Mailbox;
+      Receiver_Address : Message_Address;
+      Request_ID : Request_ID_Type;
+      Mode : Mode_Type;
+      Name : String;
+      Priority : System.Priority := System.Default_Priority)
+   with
+      Global => (In_Out => Mailboxes, Proof_In => Mailbox_Metadata),
+      Pre => true
+         and then (0 < Name'Length and Name'Length <= XDR_Size_Type'Last - 12)
+         and then Receiver_Address.Module_ID = This_Module
+         and then Receives(Receiver_Address.Module_ID, Open_Request_Msg)
+      ;
+
+   procedure Open_Reply_Encode
+      (Sender_Address : Message_Address;
+      Receiver_Address : Message_Address;
+      Request_ID : Request_ID_Type;
+      Handle : File_Handle_Type;
+      Result : out Message_Record;
+      Priority : System.Priority := System.Default_Priority)
+   with
+      Global => (Proof_In => Mailbox_Metadata),
+      Pre => true
+         and then Sender_Address.Module_ID = This_Module
+         and then Receives(Receiver_Address.Module_ID, Open_Reply_Msg),
+      Post => Message_Manager.Message_Type(Result) = Open_Reply_Msg
+       and Message_Manager.Receiver_Address(Result) = Receiver_Address;
+
+   procedure Send_Open_Reply
+      (Sender : Module_Mailbox;
+      Receiver_Address : Message_Address;
+      Request_ID : Request_ID_Type;
+      Handle : File_Handle_Type;
+      Priority : System.Priority := System.Default_Priority)
+   with
+      Global => (In_Out => Mailboxes, Proof_In => Mailbox_Metadata),
+      Pre => true
+         and then Address(Sender).Module_ID = This_Module
+         and then Receives(Receiver_Address.Module_ID, Open_Reply_Msg)
+      ;
+
+   procedure Read_Request_Encode
+      (Sender_Address : Message_Address;
+      Receiver_Address : Message_Address;
+      Request_ID : Request_ID_Type;
+      Handle : Valid_File_Handle_Type;
+      Amount : Read_Size_Type;
+      Result : out Message_Record;
+      Priority : System.Priority := System.Default_Priority)
+   with
+      Global => (Proof_In => Mailbox_Metadata),
+      Pre => true
+         and then Receiver_Address.Module_ID = This_Module
+         and then Receives(Receiver_Address.Module_ID, Read_Request_Msg),
```

Comment:
Remember to update Merc again because now Receives() only requires moduleId, not full address